java/arcs/core/storage/StorageProxy.kt

```diff
@@ -128,7 +126,7 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
                     busySendingMessagesChannel.send(false)
                 }
             }
-            .flowOn(Dispatchers.Default)
+            // TODO(yuangu-google): this should probably be flown on a different scope.
             .launchIn(scheduler.scope)
```

Comment:
The scope should always be the Scheduler's scope, so that when the scheduler is canceled, the flow's consumption is canceled. It should, however, use a different dispatcher from the scheduler - which is what `.flowOn` can afford.

Reply:
Wait a sec, launchIn() runs collect(), so if you have a hot-flow, it doesn't terminate until the channel is closed, therefore this collect() gets executed on the scheduler queue and would end up hogging it right?

Reply:
I think this is the desired effect, no? Basically here we want to keep reading msgs out of the channel and send them until the channel is closed, right?

Reply:
It shouldn't block the scheduler's thread, since `collect` suspends until there's another thing to collect... so `consumeAsFlow` *should* just queue up tasks with the scheduler thread for the `map`, and `onEach` operations.

Reply:
but collect() is an infinite loop for all intents and purposes. While it's suspending waiting for the next flow emission, will the scheduler move on to execute the next task? otherwise it will block the rest of the queue. 

The way collect() works in flows is that it will not return until no more elements are emitted, but in a hot-flow tied to a channel, this is essentially, a wait until the channel itself is closed, so collect() doesn't return immediately.